x509util/certificate_request_test.go

```diff
@@ -298,6 +303,225 @@ func TestCertificateRequest_GetCertificateRequest(t *testing.T) {
 	}
 }
 
+func TestCertificateRequest_GetCertificateRequest_challengePassword(t *testing.T) {
+	rsaPEM, err := os.ReadFile("testdata/rsa.key")
+	require.NoError(t, err)
+
+	block, _ := pem.Decode(rsaPEM)
+	rsaKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	require.NoError(t, err)
+
+	expectedPrintableString, err := os.ReadFile("testdata/challengePassword.csr")
+	require.NoError(t, err)
+	expectedUTF8String, err := os.ReadFile("testdata/challengePasswordUTF8.csr")
+	require.NoError(t, err)
+
+	tests := []struct {
+		name      string
+		cr        *CertificateRequest
+		want      []byte
+		assertion assert.ErrorAssertionFunc
+	}{
+		{"ok", &CertificateRequest{
+			Subject:            Subject{CommonName: "commonName"},
+			DNSNames:           []string{"foo.com"},
+			EmailAddresses:     []string{"root@foo.com"},
+			IPAddresses:        []net.IP{net.ParseIP("::1")},
+			URIs:               []*url.URL{{Scheme: "mailto", Opaque: "root@foo.com"}},
+			SANs:               []SubjectAlternativeName{{Type: "dns", Value: "bar.com"}},
+			ChallengePassword:  "challengePassword",
+			Extensions:         []Extension{{ID: []int{1, 2, 3, 4}, Critical: true, Value: []byte("foobar")}},
+			SignatureAlgorithm: SignatureAlgorithm(x509.SHA256WithRSA),
+			Signer:             rsaKey,
+		}, expectedPrintableString, assert.NoError},
+		{"ok UTF8String", &CertificateRequest{
```

Comment:
Nice 😄 <img width="892" alt="Screenshot 2023-12-06 at 11 06 51" src="https://github.com/smallstep/crypto/assets/1219780/88f5dd2e-6a85-4cdf-9fbd-7eaf7e93ddd1">

Reply:
It's in the spec. But I haven't checked if what we use in certificates supports it. With this PR a future PR can be the ability to parse these certificates. 

In theory, we could also use `step certificate create --csr` to create a CSR with a challengePassword using a template, but I have that disabled at the moment (`json:"-"`).